src/collocation.jl

```diff
@@ -140,3 +140,27 @@ function collocate_data(data,tpoints,kernel=TriangularKernel())
   estimated_solution = reduce(hcat,transpose.(last.(x)))
   estimated_derivative,estimated_solution
 end
+
+# FIXME: Dispatch using AbstractInterpolation not working :(
```

Comment:
what's the issue?

Reply:
I just realized that I was passing it as `CubicSpline` instead of `CubicSpline()` so it is expected to fail. However, I am wondering how to make it all fit into a common API with the Smoothened Kernel ones? Should we just have the additional `tpoints_sample` argument differentiating the functions or maybe create a wrapper around the DataInterpolations types?

Reply:
I think a wrapper over DataInterpolations types might be necessary. But also, directly defining the derivatives of its interoplations might be helpful too, since for polynomials we can just directly know the derivatives of the interpolations.

Reply:
Do you mean not using ForwardDiff for the derivatives and have those functions manually defined?

Reply:
Yes, that might make all of the differentiation much easier.